File: scala/hamming/src/main/scala/hamming.scala
class Hamming(dna1:String, dna2:String) {
    require(dna1.length == dna2.length)
    def distance =
        dna1.zip(dna2).count {
            case (a, b) => a != b
        }
}


object Hamming {
    def compute(dna1: String, dna2: String) : Int =
        new Hamming(dna1, dna2).distance
}

File: lua/hamming/hamming.lua
Hamming = {}

function Hamming.compute(str1, str2)
  local distance = 0
  for i=0, str1:len() do
    if str1:sub(i,i) ~= str2:sub(i,i) then
      distance = distance + 1
    end
  end
  return distance
end

return Hamming